src/platform/posix_io.c

```diff
@@ -590,3 +595,52 @@ oc_io_cmp oc_io_wait_single_req_for_table(oc_io_req* req, oc_file_table* table)
     }
     return (cmp);
 }
+
+oc_file_list oc_file_listdir_for_table(oc_arena* arena, oc_file directory, oc_file_table* table)
+{
+    oc_file_list list = {0};
+
+    oc_file_slot* slot = oc_file_slot_from_handle(table, directory);
+    if(slot && !slot->fatal)
+    {
+        DIR* dir = fdopendir(slot->fd);
+        if (dir)
+        {
+            struct dirent* entry = NULL;
+            while ((entry = readdir(dir)) != NULL) {
+                // skip the current/previous directory entries
+                if (!strcmp(".", entry->d_name) || !strcmp("..", entry->d_name)) {
+                    continue;
+                }
+
+                oc_file_listdir_elt* elt = oc_arena_push_type(arena, oc_file_listdir_elt);
+                oc_list_push_front(&list.list, &elt->listElt);
```

Comment:
Should we push back instead (to preserve the same order as the readdir entries)? Not sure if that matters though